### PR TITLE
Fix catalyst secondary click to match windows

### LINKED ISSUE #10464: [Win][Mac] Secondary ButtonMask odd behaviour
### Description

The [Secondary ButtonMask](https://github.com/dotnet/maui/pull/9589) PR tells us that the feature works on Windows. But I'm not having much joy with it. Bear in mind that I know that secondary won't work with "NumberOfTapsRequired".

I setup 3 Labels triggered by a single tap. Label 1 is defined to be triggered by primary. Label 2 is defined to be triggered by secondary. Label 3 is defined to be triggered by either.

Then I added three more labels. Label 4 is defined to be triggered by primary single tap. Label 5 is defined to be triggered by secondary double tap. Label 6 is defined to be triggered by either triple tap. 

#### Win results

Label 1 is triggered by primary and secondary, Label 2 and 3 work as expected.

Label 4 is triggered by a single tap (albeit in primary or secondary). Label 5 isn't triggered by any double tap (known issue). Label 6 is never triggered at all.

### Mac Catalyst results

Label 1-5 work as expected. Label 6 (triple tap with either primary or secondary) - triple tap working for primary, but secondary only requires a single tap.

### Steps to Reproduce

Run the repro case.

### Link to public reproduction project repository

https://github.com/davidbritch/dotnet-maui-issues/tree/main/TapDemo

### Version with bug

.NET 7 RC1.

### Last version that worked well

Unknown/Other

### Affected platforms

Windows and Mac Catalyst

### Affected platform versions

N/A

### Did you find any workaround?

No.

### Relevant log output

N/A.

### PR DESCRIPTION
**Pull request by PureWeen**
### Description of Change

Modify Catalyst so that secondary click only fires if the user has set the `NumberOfClicks` to 1. This is currently how it works on windows as well.  We don't want to just fire secondary click regardless of the value of ` NumberOfClicks` because that makes the world a bit weird once/if we actually implement `NumberOfClicks`

Fixes #10464

### CODE REVIEW

File: src/Controls/samples/Controls.Sample/Pages/Core/TapGestureGalleryPage.cs [hunk 1]
```diff
@@ -53,19 +53,80 @@ public TapGestureGalleryPage()
...
+			};
+			vertical.Add(horizontal2);
+
+
+			var rightClickLabel = new Label
+			{
+				Text = "Right click me¡",
+				BackgroundColor = Colors.Coral
+			};
+			var rigthClickGesture = new TapGestureRecognizer
```
Comment (samhouts) [inline src/Controls/samples/Controls.Sample/Pages/Core/TapGestureGalleryPage.cs:84]:
```suggestion
			var rightClickGesture = new TapGestureRecognizer
```

File: src/Controls/samples/Controls.Sample/Pages/Core/TapGestureGalleryPage.cs [hunk 2]
```diff
@@ -53,19 +53,80 @@ public TapGestureGalleryPage()
...
+			{
+				Command = TapCommand,
+				CommandParameter = Colors.Coral,
+				NumberOfTapsRequired = 1,
+				Buttons = ButtonsMask.Secondary
+			};
+			rigthClickGesture.Tapped += OnTapped;
```
Comment (samhouts) [inline src/Controls/samples/Controls.Sample/Pages/Core/TapGestureGalleryPage.cs:91]:
```suggestion
			rightClickGesture.Tapped += OnTapped;
```

File: src/Controls/samples/Controls.Sample/Pages/Core/TapGestureGalleryPage.cs [hunk 3]
```diff
@@ -53,19 +53,80 @@ public TapGestureGalleryPage()
...
+			rightClickLabel.GestureRecognizers.Add(rigthClickGesture);
```
Comment (samhouts) [inline src/Controls/samples/Controls.Sample/Pages/Core/TapGestureGalleryPage.cs:92]:
```suggestion
			rightClickLabel.GestureRecognizers.Add(rightClickGesture);
```

File: src/Controls/samples/Controls.Sample/Pages/Core/TapGestureGalleryPage.cs [hunk 4]
```diff
@@ -53,19 +53,80 @@ public TapGestureGalleryPage()
...
+			horizontal2.Children.Add(rightClickLabel);
+
+			var rightorLeftClickLabel = new Label
+			{
+				Text = "Right or Left click me¡",
+				BackgroundColor = Colors.Green
+			};
+			var rigthOrLeftClickGesture = new TapGestureRecognizer
```
Comment (samhouts) [inline src/Controls/samples/Controls.Sample/Pages/Core/TapGestureGalleryPage.cs:100]:
```suggestion
			var rightOrLeftClickGesture = new TapGestureRecognizer
```

File: src/Controls/samples/Controls.Sample/Pages/Core/TapGestureGalleryPage.cs [hunk 5]
```diff
@@ -53,19 +53,80 @@ public TapGestureGalleryPage()
...
+			{
+				Command = TapCommand,
+				CommandParameter = Colors.Green,
+				NumberOfTapsRequired = 1,
+				Buttons = ButtonsMask.Secondary | ButtonsMask.Primary
+			};
+
+			rigthOrLeftClickGesture.Tapped += OnTapped;
+			rightorLeftClickLabel.GestureRecognizers.Add(rigthOrLeftClickGesture);
```
Comment (samhouts) [inline src/Controls/samples/Controls.Sample/Pages/Core/TapGestureGalleryPage.cs:109]:
```suggestion
			rightOrLeftClickGesture.Tapped += OnTapped;
			rightorLeftClickLabel.GestureRecognizers.Add(rightOrLeftClickGesture);
```

File: src/Controls/samples/Controls.Sample/Pages/Core/TapGestureGalleryPage.cs [hunk 6]
```diff
@@ -53,19 +53,80 @@ public TapGestureGalleryPage()
...
+			horizontal2.Children.Add(rightorLeftClickLabel);
+
+			var rightorLeftClickLabel2Taps = new Label
+			{
+				Text = "Double Click Me! ",
+				BackgroundColor = Colors.Green,
+			};
+
+			var rigthOrLeftClickGesture2Taps = new TapGestureRecognizer
```
Comment (samhouts) [inline src/Controls/samples/Controls.Sample/Pages/Core/TapGestureGalleryPage.cs:118]:
```suggestion
			var rightOrLeftClickGesture2Taps = new TapGestureRecognizer
```

File: src/Controls/samples/Controls.Sample/Pages/Core/TapGestureGalleryPage.cs [hunk 7]
```diff
@@ -53,19 +53,80 @@ public TapGestureGalleryPage()
...
+			{
+				Command = TapCommand,
+				CommandParameter = Colors.Green,
+				NumberOfTapsRequired = 2,
+				Buttons = ButtonsMask.Secondary | ButtonsMask.Primary
+			};
+
+			rigthOrLeftClickGesture2Taps.Tapped += OnTapped;
+			rightorLeftClickLabel2Taps.GestureRecognizers.Add(rigthOrLeftClickGesture2Taps);
```
Comment (samhouts) [inline src/Controls/samples/Controls.Sample/Pages/Core/TapGestureGalleryPage.cs:127]:
```suggestion
			rightOrLeftClickGesture2Taps.Tapped += OnTapped;
			rightorLeftClickLabel2Taps.GestureRecognizers.Add(rightOrLeftClickGesture2Taps);
```

**Review by samhouts — COMMENTED** [review]
phew, glad i was here

File: src/Controls/src/Core/Platform/GestureManager/GestureManager.Windows.cs [hunk 8]
```diff
@@ -565,11 +565,13 @@ bool ProcessGestureRecognizers(IEnumerable<TapGestureRecognizer>? tapGestures)
 
 			bool ValidateGesture(TapGestureRecognizer g)
 			{
-				if (e is RightTappedRoutedEventArgs &&
-					(g.Buttons & ButtonsMask.Secondary) == ButtonsMask.Secondary)
+				if (e is RightTappedRoutedEventArgs)
 				{
 					// Currently we only support single right clicks on WinUI
```
Comment (rachelkang) [inline src/Controls/src/Core/Platform/GestureManager/GestureManager.Windows.cs:570]:
```suggestion
					// Currently we only support single right clicks
```

**Review by rachelkang — APPROVED** [review]

**Review by rmarinho — APPROVED** [review]

**Pull request merged**